src/components/playback/playbackmanager.js

```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
 
     function getIntros(firstItem, apiClient, options) {
 
-        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !userSettings.enableCinemaMode()) {
+        var unwatchedOnly = userSettings.enableCinemaTrailersUnseenOnly();
+        var trailerCount = userSettings.cinemaModeTrailerCount();
+        var enableCinema = userSettings.enableCinemaMode();
+
+        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !enableCinema || trailerCount == 0) {
             return Promise.resolve({
                 Items: []
             });
         }
 
-        return apiClient.getIntros(firstItem.Id).then(function (result) {
+        var fetchOptions = {
+            Fields: 'RemoteTrailers,LocalTrailerCount,ServerID',
+            Filters: unwatchedOnly ? 'IsUnPlayed' : '',
+            HasTrailer: true,
+            IncludeItemTypes: 'Movie',
+            Limit: trailerCount,
+            Recursive: true,
+            SortBy: 'Random'
+        };
+
+        function getRemoteTrailers(item) {
+            var remoteTrailers = item.RemoteTrailers || [];
+
+            if (!remoteTrailers.length) {
+                return [];
+            }
+
+            return remoteTrailers.map(function (t) {
+                return {
+                    Name: t.Name || (item.Name + ' Trailer'),
+                    Url: t.Url,
+                    MediaType: 'Video',
+                    Type: 'Trailer',
+                    ServerId: apiClient.serverId()
+                };
+            });
+        }
+
+        function getAllTrailers(item, remoteTrailers) {
+            if (item.LocalTrailerCount) {
+                return apiClient.getLocalTrailers(apiClient.getCurrentUserId(), item.Id).then(function (trailers) {
+                    trailers.push.apply(trailers, remoteTrailers);
```

Comment:
any reason _not_ to call `trailers.push(remoteTrailers)`?

Reply:
We should push elements of `remoteTrailers`, not the array itself, i.e merge arrays